test/e2e/utils.go

```diff
@@ -328,22 +328,45 @@ func toUnstructured(y string) interface{} {
 	return u
 }
 
-func bridgeVlansAtNodes() []string {
-	outputs, _ := runAtNodes("sudo", "bridge", "-j", "vlan", "show")
-	return outputs
+func bridgeVlansAtNode(node string) (string, error) {
+	return run(node, "sudo", "bridge", "-j", "vlan", "show")
 }
 
-func hasVlans(bridgeVlans string, connection string, minVlan int, maxVlan int) {
+func hasVlans(node string, connection string, minVlan int, maxVlan int) AsyncAssertion {
```

Comment:
We always use this in combination with Should(Succeed()). Would you mind just checking it inside the function (maybe rename it it checkVlans). I'd like to do some tiny refactoring on tests code and get rid of returning AsyncAssertion. It is killing my brain. Even worse when I have to explain how it works. I think our helper function should either return some basic Go value or do the whole assertion. This is on a mid way, but I don't think it makes the code more readable.

Feel free to argue. I don't believe AsyncAssertion gives us much value with `hasVlans(node, "bond1", 2, 4094).Should(Succeed())`. `hasVlans` either succeeds or not. If you want it for Eventually, return error, if not, assert inside the function.

Similar to vlansCardinality.

Reply:
There is one issue, there is no concept of "EventualyWithOffset" so if we do the "Should" inside the function the the line of code that appears on error is the function not the test itself, returning AsyncAssertion is kind of idiom we can adopt here one option is to state it in the function with something like "eventuallyHasVlans" so we know what we are testing.